src/nifgen/system_tests/test_system_nifgen.py

```diff
@@ -0,0 +1,55 @@
+import nifgen
+import pytest
+
+
+@pytest.fixture(scope='function')
+def session():
+    with nifgen.Session('', False, False, 'Simulate=1, DriverSetup=Model:5433 (2CH);BoardType:PXIe') as simulated_session:
+        yield simulated_session
+
+
+def test_self_test(session):
+    result, message = session.self_test()
+    assert result == 0
+    assert message == 'Self test passed'
+
+
+def test_get_attribute_string(session):
+    model = session.instrument_model
+    assert model == 'NI PXIe-5433 (2CH)'
+
+
+def test_error_message(session):
+    # Calling the private function directly, as _get_error_message() only gets called when you have an invalid session,
+    # and there is no good way for us to invalidate a simulated session.
+    message = session._error_message(-1074135027)
+    assert message.find('Attribute is read-only.') != -1
+
+
+def test_get_error(session):
+    try:
+        session.instrument_model = ''
+        assert False
+    except nifgen.Error as e:
+        assert e.code == -1074135027  # Error : Attribute is read-only.
+        assert e.description.find('Attribute is read-only.') != -1
+
+
+def test_method_get_self_cal_supported(session):
+    assert session.get_self_cal_supported() in [True, False]
+
+
+def test_get_self_cal_last_date_and_time(session):
```

Comment:
If you are not going to use the session created in the "session" fixture. Remove it.

aka:
test_get_self_cal_last_date_and_time():